error/s2n_errno.c

```diff
@@ -103,6 +103,7 @@ static const char *no_such_error = "Internal s2n error";
     ERR_ENTRY(S2N_ERR_CERT_INVALID, "Certificate is invalid") \
     ERR_ENTRY(S2N_ERR_CERT_MAX_CHAIN_DEPTH_EXCEEDED, "The maximum certificate chain depth has been exceeded") \
     ERR_ENTRY(S2N_ERR_CERT_REJECTED, "Certificate failed custom application validation") \
+    ERR_ENTRY(S2N_ERR_SECURITY_POLICY_INCOMPATIBLE_CERT, "Certificate is not permitted by the certificate preferences of the security policy") \
```

Comment:
Are we committed to keeping this a local only error message? My thinking was that it would be useful to have a single "your security policy certificate preferences caused errors" message, regardless of whether that was local or peer based.

Reply:
My attempt to summarize the comments so far.

Yes we are committed to keeping this a local only error message. While it is true that it would be useful to also return a more specific error message for the validation failure against peer certs, we might as well make that errno a peer specific message.